### PR TITLE
Unset compiler envvars in unit tests.

### PR DESCRIPTION
**Pull request by jpakkane**

### CODE REVIEW

File: run_unittests.py [hunk 1]
```diff
@@ -1556,5 +1556,15 @@ def test_subdir(self):
...
 
 
+def unset_envs():
+    # For unit tests we must fully control all commend lines
+    # so that there are no unexpected changes coming from the
+    # environment, for example when doing a package build.
+    varnames = ['CFLAGS', 'CXXFLAGS', 'CPPFLAGS']
+    for v in varnames:
+        if v in os.environ:
+            del os.environ[b]
```
Comment (ignatenkobrain) [inline run_unittests.py]: __[flake8]__
* [F821] undefined name 'b'

File: run_unittests.py [hunk 2]
```diff
@@ -1556,5 +1556,15 @@ def test_subdir(self):
         self.assertEqual(s2, self.read_contents('sub2/meson.build'))
 
 
+def unset_envs():
+    # For unit tests we must fully control all commend lines
```
Comment (QuLogic) [inline run_unittests.py]: command

File: run_unittests.py [hunk 3]
```diff
@@ -1556,5 +1556,15 @@ def test_subdir(self):
...
+    # so that there are no unexpected changes coming from the
+    # environment, for example when doing a package build.
+    varnames = ['CFLAGS', 'CXXFLAGS', 'CPPFLAGS']
```
Comment (QuLogic) [inline run_unittests.py]: What about `FFLAGS`, `DFLAGS`, `VALAFLAGS` and everything else [out of here](https://github.com/mesonbuild/meson/blob/aa486f5bb2b962ee6f5f65cabe05bb33f92ebead/mesonbuild/environment.py#L813)? `LDFLAGS`, too?

**Comment by jpakkane** [comment]
Updated.

**Comment by tp-m** [comment]
I think it would be nice if the commit message provided an explanation *why* this has to be done, for posterity :)

**Comment by jpakkane** [comment]
The function comment already does that.

**Pull request merged**